docs/GoogleSsoRegistration.md

```diff
@@ -0,0 +1,70 @@
+# Google Authentication Registration
+
+In order to allow users to log into the Personal Data Server web application
+using their Google account, the Personal Data Server must to registered with
+Google.  The result of the registration process is that Google generates a
+client id and secret for the Personal Data Server to use that for its single
+sign-on workflow.  The following provides information regarding the Google
+registration process.
+
+Note that this outlines what is required for a test server under development.
+The steps for registering a production server are similar but with some
+differences; and involve additional steps such as submitting an application to
+Google for [verification](https://support.google.com/cloud/answer/9110914).
+Where applicable, the instructions use the phrase "for development" to highlight
+the difference.
+
+1. Create or select a Project in the [Google Cloud Console](https://console.cloud.google.com).
+2. Configure authentication consent using the "OAuth Consent Screen" page in the
+   "APIs and Services" section of the Console.
+3. For development, for "User Type", choose "External".  Submitting the form
+   will start an "Edit app registration" step-by-step process for configuring
+   the "OAuth consent screen".  At step one:
+    * For developement, fill in the "App information" with something
+      meaningful to you and other developers.
+    * For development, under "App Domain", use a local host url with http
+      for the "Application home page". The port is up to the developer.  An
+      example is `http://localhost:3000`.
+    * For development, under "Application privacy policy link", use the
+      proposed privacy policy at the root of the project: `http://localhost:3000/PRIVACY_POLICY.md`.
+      This file is located in the source code root directory.
+    * For development, Under "Application privacy policy link", use the
+      dummy service agreement at the root of the project: `http://localhost:3000/TERMS_OF_SERVICE.md`, also located in the root folder of
+      the source.
+    * For "Authorized domains" use `ocadu.ca`.
```

Comment:
Is this necessary? What use is made of this domain info?

Reply:
After reviewing Google's documentation on this, this not necessary for development, but will be for production.

Google says, "If you added Authorized JavaScript Origins and Authorized Redirect URIs when you created your credentials, they will also appear here."  The authorized origins and redirect URIs are set at step 8 and, for development, are `localhost` URIs.  The `localhost` does not automatically appear for the "Authorized Domains", and I suspect the reason is that `localhost` does not qualify as an official domain.

Google also says, "When a domain is used on the consent screen or in an OAuth client’s configuration, it must be pre-registered here".  That will be relevant to a production system, since the URIs for production will have actual domains.

Finally, as a quick test, I removed any authorized domain from my instance such that the field was blank, and ran the Google SSO for the development version of the Personal Data server, and nothing untoward occurred.